llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -5034,7 +5034,8 @@ class TargetLowering : public TargetLoweringBase {
   /// returns a register number of 0 and a null register class pointer.
   virtual std::pair<unsigned, const TargetRegisterClass *>
   getRegForInlineAsmConstraint(const TargetRegisterInfo *TRI,
-                               StringRef Constraint, MVT VT) const;
+                               StringRef Constraint, MVT VT,
+                               std::string &ErrMsg) const;
```

Comment:
Maybe SmallStringImpl&?

I don't really see how you can do much better than that; an error message is, ultimately, just a string, given we don't support translation or anything like that.

Reply:
Error or StringError?

Reply:
I like the idea of `SmallString`. `Error` and `StringError`, they consist of `std::string` and additionally have a return code that we don't have use of.

Ok, since there aren't strong points for using something like `Expected`, I'll continue with returning an error message by reference.